ios/CameraView+TakePhoto.swift

```diff
@@ -85,28 +120,39 @@ extension CameraView {
           promise.reject(error: .parameter(.invalid(unionName: "QualityPrioritization", receivedValue: qualityPrioritization)))
           return
         }
-        photoSettings.photoQualityPrioritization = photoQualityPrioritization
+        photoSettings!.photoQualityPrioritization = photoQualityPrioritization
       }
 
       // red-eye reduction
       if #available(iOS 12.0, *), let autoRedEyeReduction = options["enableAutoRedEyeReduction"] as? Bool {
-        photoSettings.isAutoRedEyeReductionEnabled = autoRedEyeReduction
+        photoSettings!.isAutoRedEyeReductionEnabled = autoRedEyeReduction
       }
 
       // stabilization
       if let enableAutoStabilization = options["enableAutoStabilization"] as? Bool {
-        photoSettings.isAutoStillImageStabilizationEnabled = enableAutoStabilization
+        photoSettings!.isAutoStillImageStabilizationEnabled = enableAutoStabilization
       }
 
       // distortion correction
       if #available(iOS 14.1, *), let enableAutoDistortionCorrection = options["enableAutoDistortionCorrection"] as? Bool {
-        photoSettings.isAutoContentAwareDistortionCorrectionEnabled = enableAutoDistortionCorrection
+        photoSettings!.isAutoContentAwareDistortionCorrectionEnabled = enableAutoDistortionCorrection
       }
 
-      photoOutput.capturePhoto(with: photoSettings, delegate: PhotoCaptureDelegate(promise: promise))
-
+      if self.enableRawCapture?.boolValue == true {
+        // Create a delegate to monitor the capture process.
+        let delegate = RAWCaptureDelegate(promise: promise)
+        self.captureDelegates[photoSettings!.uniqueID] = delegate
+        // Remove the delegate reference when it finishes its processing.
+        delegate.didFinish = {
+          self.captureDelegates[photoSettings!.uniqueID] = nil
+        }
+        // Tell the output to capture the photo.
+        photoOutput.capturePhoto(with: photoSettings!, delegate: delegate)
+      } else {
+        photoOutput.capturePhoto(with: photoSettings!, delegate: PhotoCaptureDelegate(promise: promise))
+      }
```

Comment:
This as well - we should probably extend (or subclass) `PhotoCaptureDelegate` to also be able to take RAW photos instead of adding this nested branch here. Or we make it a protocol and have two implementations, one for RAW photos one for "normal" photos.